File: 14-longest-common-prefix/14-longest-common-prefix.cpp
#include<algorithm>

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.size() == 0){
            return "";
        }else if(strs.size() ==1){
            return strs[0];
        }else {
            sort(strs.begin(), strs.end());
            int min_ = min(strs[0].size(), strs[strs.size() -1].size());
            int i = 0;
            while(i <min_ && strs[0][i] == strs[strs.size() -1][i]){
                i++;
            };
        return strs[0].substr(0, i);
           
    };
}; 
};

File: 14-longest-common-prefix/14-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
   if(strs.length === 0 || strs.includes("")){
       return "";
   }else if(strs.length ===1){
       return strs[0];
   }else{
        let out = strs[0][0];
       let index = 1;
      while(out.length <= strs[0].length){
           for(let i = 1; i<strs.length; i++){
           if(out !== strs[i].slice(0,index)){
               return out.slice(0, index-1);
           }
       }
        index +=1;
        out += strs[0][index-1];
      }
       return out.slice(0, index-1);
   }
    
};